feat: added bindings from vim index

mly32 opened the pull request:
### Summary

The built-in bindings described in which-key aren't exhaustive, and I would like them to be. This PR changes the preset keybindings. Namely, it modifies `plugins.presets.init` and adds `plugins.presets.extra` in replacement of `plugins.presets.misc`.

This is **not** ready to be merged (see [Issues](###Issues)).

Some Unicode characters are used to indicate some information about the commands, just like how `+` is used to indicate a prefix by default:
- `↶`: command can be undone,
- `✝`: command can have a register prefix,
- `§`: command expects a motion (aka command is an operator),
- `⤨`: command is a motion.

For example, `["c"] = "↶,✝,§ delete text [into register x] and start insert"`.

The current configuration:
- excludes mouse binds,
- excludes the `<C-...>` binds in windows mode that are all "same as",
- popup binds (see `extra.insert`) are not shown (I'm not sure how they would be),
- does not show symbols/indicators for keybindings that expect a char (ex. `f``),
- descriptions are not completely true to the source (vim index).

### Notes

There can be the following additional configuration options (not implemented):

- choose a naming style (e.g., `<c-s-o>` vs. `<C-S-o>` vs. `<C-S-O>`).
- have a field during `setup` to add a list of renames for the descriptions for the preset built-in keybindings.
- have options to replace `↶, ✝, §, ⤨` icons.
- like how there is the `operators` option in the setup, there could also be a `motions` option.
- have a toggle for showing or hiding:
  - "same as" keybindings (e.g., `h` = `<Left>` = `<C-h>` = `<BS>`),
  - mouse keybindings (e.g. scroll wheel and clicks).
- a check of the `~` command for different a description and interaction based on the state of `tildeop` (there are other commands that also have varied actions based on settings).

I currently do not have time in my schedule to make these necessary improvements, but the current implementation works well enough for me. I thought I would start this branch so that other contributors can make use of it.

This was the setup I used:
```Lua
wk.setup({
  operators = {},
  layout = {height = {min = 4, max = 32}, width = {min = 20, max = 80}},
  triggers_blacklist = {}
})
```

### Issues

- The keybindings make the popup window LONG!!!
- There should be more indicators for commands expecting additional input (either a prefix or suffix) and what type of input (number, character, motion). If possible, a plugin can maybe have a more complex lexer and display of keybindings so that numbers and character ranges (e.g., `{a-zA-Z0-9}`) can be used as a prefix.
- When I ran `checkhealth` I got conflicting keymap warnings, but I was unable to resolve them.
- Some descriptions are too long and I've already changed some of the descriptions (unfortunately, I did not keep track of which was changed). There needs to be a way to shorten the descriptions consistently.
- When going down a chain of commands (e.g., `ciw`), if I scroll down during one layer and go into the next layer, the scroll is not reset to the top.

folke commented:
Hi! Thank you for doing this. Some remarks:

* the lists would be probably too long for most users. Right now, the preset is based of a couple of quickcheat lists for vim to get the most used keybindings.
* would be best if all the extra bindings would be under an options `opts.extra` and keep the `misc` bindings as is
* the default value for `opts.extra` for the `preset` plugin should be `false`
* I also did some work on manually cleaning the descriptions that came from the help (like removing `"`, etc.)
* the way which-key works right now, is a bit hacky to be honest. It works as is for most of the use cases and there is currently no better way to implement this, but it's definitely fragile in some ways. Adding a lot of new operators could possibly break keybindings in unexpected places :). But again, if it's under `opts.extra` I'm fine with that.
* I'll fix the scrolling. I know what's happening there and recently implemented a better solution for compe that I still need to port to which-key

Thanks again for the effort. I'll keep this open for now, but won't merge it as is. I also don't have the time to work on this, and it's also not something I would use myself, but I can imagine new users might be interested in more presets.

mly32 commented:
Thanks for the fast reply. I think I'll move the changes to `opts.extra` like you said, clean up the descriptions and the keymap conflicts, and try to get a more comprehensive list of keybindings that were not included in `extra`. I'll get more time later this week and hopefully complete it then.

mly32 commented:
I had an issue merging and somehow ended up creating a new PR for this. Apologies.

folke commented:
no problem :)

I did notice you've been reformatting most source files. This repo uses `stylua` and has a proper configuration for this. Please don't use anything else, since in its current state, this PR is impossible for me to review.

mly32 commented:
Thanks for the update. I've switched to `style, which drastically reduced the number of changed lines. I also updated the summary with more information about what I did. It seems like I was able to get this PR working again, so I ended up deleting the other PR.

:neutral_face: sorry for the confusion.

mly32 commented:
@folke I haven't made any updates since you were away. Would you look over what I have so I can make any required changes? Thanks.

tmillr commented:
Nice this looks awesome! As a new which-key user, I was wondering myself why the builtin "keymap" listings weren't complete/exhaustive and I'd really love to see this implemented. I also don't like how `v` stalls and opens up the popup because it just gets in the way (especially if you're on a laptop with a smaller screen) and feels unintuitive. Plus, `v` isn't an operator (or is it?), and without which-key `v` doesn't wait for a motion.

textzenith commented:
Thanks,

I couldn't figure out why the keymap listings are so incomplete, either.

For someone using Vim for the very first time, the suggestions are good. For long-term users this is a frustration.

Rather than enabing "extra" mode, this should be about disabling "noob" mode!

As for the Greek characters, this is a surefire way to make Neovim _completely, bewilideringly confusing and unusable_ to first-time users. However, there's the grain of a very idea here in educating users which commands are motions, register- or mark- based, etc. etc.
